OpenSim/Sandbox/MatlabScripts/createActuatorsFile.m

```diff
@@ -0,0 +1,165 @@
+[filename, pathname] = uigetfile('*.osim', 'Select an OpenSim Model File');
+
+% get the model path
+modelFilePath = fullfile(pathname,filename);
+
+% Generate an instance of the model
+model = Model(modelFilePath);
+
+% Get the number of coordinates  and a handle to the coordainte set
+nCoord = model.getCoordinateSet.getSize();
+coordSet = model.getCoordinateSet();
+
+% Evaluate the ground body and get the mass center
+groundBodyName = model.getGround().getName();
+groundJoint = model.getJointSet.get(0);
+
+% Create some empty vec3's for later.
+massCenter = Vec3();
+axisValues = Vec3();
+
+% Create an empty Force set
+forceSet = ForceSet();
+% get the state
+state = model.initSystem();
+
+optimalForce = 1;
+
+%% Start going through the coordinates, creating an actuator for each
+for iCoord = 0 : nCoord - 1
+
+    % get a reference to the current coordinate
+    coordinate = coordSet.get(iCoord);
+    % If the coordinate is locked don't add an actuator.
+    if coordinate.getLocked(state)
+        continue
+    end
+    % If the coordinate is prescribed, don't add an actuator.
+    if coordinate.isPrescribed(state)
+        continue
+    end
+    % If the coodinate is constrained, don't add an actuator
+    if coordinate.isConstrained(state)
+        continue
+    end
+
+    % get the joint, parent and child names for the coordiante
+    joint = coordinate.getJoint;
+    parentName = joint.getParentFrame().getName(); 
+    childName = joint.getChildFrame().getName();
+
+    % If the coordinates parent body is connected to ground, we need to
+    % add residual actuators (torque or point).
+    if strmatch(parentName, model.getGround.getName() )
+
+        % if the joint type is custom or free
+        if strcmp(joint.getConcreteClassName(), 'CustomJoint') | strcmp(joint.getConcreteClassName(), 'FreeJoint')
+               % get the coordainte motion type
+               motion = char(coordinate.getMotionType);
+               % to get the axis value for the coordiante, we need to drill
+               % down into the coordinate transform axis
+               eval(['concreteJoint = ' char(joint.getConcreteClassName()) '.safeDownCast(joint);'])
+               sptr = concreteJoint.getSpatialTransform;
+               for ip = 0 : 5
+                  if strcmp(char(sptr.getCoordinateNames().get(ip)), char(coordinate.getName))
+                        sptr.getTransformAxis(ip).getAxis(axisValues);
+                        break
+                  end
+               end
+
+
+               % Build a torque actuator for a rotational coordinate
+               if strcmp(motion, 'Rotational')
+                   newActuator = TorqueActuator(joint.getParentFrame(),...
+                                         joint.getParentFrame(),...
+                                         axisValues,...
+                                         1);
+    
+               % Build a point actuator for a translational coordainte.
+               else 
```

Comment:
The comment doesn't match the code. There are actually 4 MotionTypes, 3 of which might be encountered here (Rotational, Translational, and Coupled). The `else` should be an `elseif` and presumably Coupled should be handled as well.